command/genesis/predeploy/params.go

```diff
@@ -32,28 +36,37 @@ var (
 )
 
 type predeployParams struct {
-	addressRaw  string
-	genesisPath string
+	addressRaw      string
+	genesisPath     string
+	deployerAddrRaw string
+
+	address       types.Address
+	deployerAddr  types.Address
+	artifactsName string
+	artifactsPath string
 
-	address         types.Address
-	artifactsPath   string
 	constructorArgs []string
 
-	genesisConfig *chain.Chain
+	genesisConfig    *chain.Chain
+	contractArtifact *artifact.Artifact
 }
 
 func (p *predeployParams) getRequiredFlags() []string {
 	return []string{
 		predeployAddressFlag,
-		artifactsPathFlag,
+		artifactsNameFlag,
```

Comment:
I'd still require artifactsPathFlag as it was before. Providing artifact via the name is optional convenience.

```suggestion
		artifactsPathFlag,
```